pkg/config/virtio.go

```diff
@@ -181,6 +220,165 @@ func (dev *VirtioSerial) FromOptions(options []option) error {
 	return dev.validate()
 }
 
+// VirtioInputNew creates a new input device for the virtual machine.
+// The inputType parameter is the type of virtio-input device that will be added
+// to the machine.
+func VirtioInputNew(inputType string) (VirtioDevice, error) {
+	if inputType != VirtioInputPointingDevice && inputType != VirtioInputKeyboardDevice {
+		return nil, fmt.Errorf("Unknown option for virtio-input devices: %s", inputType)
+	}
+
+	return &VirtioInput{
+		InputType: inputType,
+	}, nil
+}
+
+func (dev *VirtioInput) validate() error {
+	if dev.InputType != VirtioInputPointingDevice && dev.InputType != VirtioInputKeyboardDevice {
+		return fmt.Errorf("Unknown option for virtio-input devices: %s", dev.InputType)
+	}
+
+	return nil
+}
+
+func (dev *VirtioInput) ToCmdLine() ([]string, error) {
+	if err := dev.validate(); err != nil {
+		return nil, err
+	}
+
+	return []string{"--device", fmt.Sprintf("virtio-input,%s", dev.InputType)}, nil
+}
+
+func (dev *VirtioInput) FromOptions(options []option) error {
+	for _, option := range options {
+		if option.key != VirtioInputPointingDevice && option.key != VirtioInputKeyboardDevice {
+			return fmt.Errorf("Unknown option for virtio-input devices: %s", option.key)
+		}
+		if option.value != "" {
+			return fmt.Errorf(fmt.Sprintf("Unexpected value for virtio-input %s option: %s", option.key, option.value))
+		}
+		dev.InputType = option.key
+	}
+	return dev.validate()
+}
+
+// VirtioGPUNew creates a new gpu device for the virtual machine.
+// The device parameter is the virtio-gpu device that will be added to the machine
+func VirtioGPUNew(device string) (VirtioDevice, error) {
```

Comment:
I'm not fully clear why this `device` string is required? This does not seem to be used in `Code-Hex/vz`, and it must be `display` in vfkit code?